porcupine/plugins/anchors.py

```diff
@@ -0,0 +1,173 @@
+from __future__ import annotations
+import tkinter
+import time
+from typing import List
+from porcupine import tabs, get_tab_manager
+from porcupine.plugins.linenumbers import LineNumbers
+
+# Dependent on code from linenumbers.py
+setup_after = ["linenumbers"]
+
+
+class AnchorManager:
+    def __init__(self, tab_textwidget: tkinter.Text, linenumbers: tkinter.Canvas) -> None:
+        self.tab_textwidget = tab_textwidget
+        self.linenumbers = linenumbers
+
+        self.anchor_symbol = "¶"
+
+        # tkinter have default marks such as "insert", "current", "tkinter::anchor1"
+        self.custom_anchor_prefix = "anchor_"
+
+        linenumbers.bind("<<Updated>>", self.update_linenumbers, add=True)
+
+    def _get_anchors(self) -> List:
+        return [
+            anchor
+            for anchor in self.tab_textwidget.mark_names()
+            if anchor.startswith(self.custom_anchor_prefix)
+        ]
+
+    def _get_cursor_index(self):
+        return self.tab_textwidget.index("insert linestart")
+
+    def toggle_on_off(self, event: tkinter.Event[tkinter.Misc]) -> None:
+
+        self.prevent_duplicate_anchors()
+
+        cursor_index = self._get_cursor_index()
+        for anchor in self.tab_textwidget.mark_names():
+            if (
+                anchor.startswith(self.custom_anchor_prefix)
+                and self.tab_textwidget.index(anchor) == cursor_index
+            ):
+                self.tab_textwidget.mark_unset(anchor)
+                self.linenumbers.do_update()
+                return
+        self.tab_textwidget.mark_set(
+            self.custom_anchor_prefix + str(time.time()), "insert linestart"
+        )
+
+        self.linenumbers.do_update()
+
+    def jump_to_next(self, event: tkinter.Event[tkinter.Misc]) -> None:
+        cursor_row = self._get_cursor_index().split(".")[0]
+        anchor_list = self._get_anchors()
+        anchor_rows = reversed(
+            sorted(
+                [
+                    int(self.tab_textwidget.index(anchorpoint).split(".")[0])
+                    for anchorpoint in anchor_list
+                ]
+            )
+        )
+
+        next_anchor_row = 0
+        for anchor_row in anchor_rows:
+            if anchor_row > int(cursor_row):
+                next_anchor_row = anchor_row
+            else:
+                break
```

Comment:
This logic is a bit difficult to follow. How about the following:

```suggestion
        rows_after_cursor = [n for n in anchor_rows if n > int(cursor_row)]
        if rows_after_cursor:
            next_anchor_row = min(rows_after_cursor)
        else:
            next_anchor_row = 0
```

This leads to other simplifications:
- Because `min()` does not care about the order of the list given to it, you no longer need `sorted()` or `reversed()`.
- You can `return` instead of `next_anchor_row = 0`.

Reply:
Ah this makes sense! I need to learn to use min / max more!